homeassistant/components/light/futurenow.py

```diff
@@ -0,0 +1,134 @@
+    vol.Required(CONF_PORT): cv.string,
+    vol.Required(CONF_DEVICES): {cv.string: DEVICE_SCHEMA},
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the light platform for each FutureNow unit."""
+    lights = []
+    for channel, device_config in config[CONF_DEVICES].items():
+        device = {}
+        device['name'] = device_config[CONF_NAME]
+        device['dimmable'] = True if 'dimmable' in device_config else False
+        device['channel'] = channel
+        device['driver'] = config[CONF_DRIVER]
+        device['host'] = config[CONF_HOST]
+        device['port'] = config[CONF_PORT]
+        lights.append(FutureNowLight(device))
+
+    add_devices(lights)
+
+
+def to_futurenow_level(level):
+    """Convert the given HASS light level (0-255) to FutureNow (0-100)."""
+    return int((level * 100) / 255)
+
+
+def to_hass_level(level):
+    """Convert the given FutureNow (0-100) light level to HASS (0-255)."""
+    return int((level * 255) / 100)
+
+
+class FutureNowLight(Light):
+    """Representation of an FutureNow light."""
+
+    def __init__(self, device):
+        """Initialize the light."""
+        import pyfnip
+
+        self._name = device['name']
+        self._dimmable = device['dimmable']
+        self._channel = device['channel']
+        self._brightness = None
+        self._state = None
+
+        if device['driver'] == CONF_DRIVER_FNIP6X10AD:
+            self._light = pyfnip.FNIP6x2adOutput(device['host'],
+                                                 device['port'],
+                                                 self._channel)
+        if device['driver'] == CONF_DRIVER_FNIP8X10A:
+            self._light = pyfnip.FNIP8x10aOutput(device['host'],
+                                                 device['port'],
+                                                 self._channel)
+        # Get actual state of light.
+        self.update()
+
+    @property
+    def name(self):
+        """Return the name of the device if any."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return true if device is on."""
+        return self._state
+
+    @property
+    def brightness(self):
+        """Return the brightness of this light between 0..255."""
+        return self._brightness
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        if self._dimmable is True:
+            return SUPPORT_BRIGHTNESS
+        return 0
+
+    def turn_on(self, **kwargs):
+        """Turn the light on."""
+        level = kwargs.get(ATTR_BRIGHTNESS, 255) if self._dimmable else 255
+        self._light.turn_on(to_futurenow_level(level))
+
+    def turn_off(self, **kwargs):
+        """Turn the light off."""
+        self._light.turn_off()
+
+    def update(self):
+        """Fetch new state data for this light."""
+        # Delay a bit until state change has fully finished."
+        time.sleep(.500)
```

Comment:
Sleeping is not allowed in sync methods. What is the symptom that we're trying to mitigate with sleeping?

Reply:
FutureNow dimmer unit has a built-in fade-in effect on state changes. If status is polled immediately after brightness change command, the change might still be underway. This is seen in the UI by first receiving an incorrect brightness, but corrected in next poll.

Reply:
If it isn't possible to get correct feedback faster, we can set the brightness instance attribute to the updated value in the actuator method, to avoid the state jumping.

Reply:
Wait, that won't work either since `update` will overwrite that.

Reply:
An easier alternative could be to change the actuator methods to async coroutines. Then we can use async sleep without penalty to any thread pool. I think that might be better since we only need a short sleep time.

We then schedule the library call on the executor with `hass.async_add_executor_job`.

Reply:
I decided to try with selective skipping of the update.. It seems to work ok and the UI receives feedback of the state change immediately. Would this be good to go?